### DISCUSSION TITLE
Error establishing a database connection

### DISCUSSION THREAD

**Discussion by rahmaazs** [leokhoa/laragon#446] [Q&A]
I was trying to work on my project, but when i clicked "Start All" and visit the site, it appears "Error establishing a database connection" like this :
![Screenshot (540)](https://user-images.githubusercontent.com/113582506/226223053-8b82678d-b838-4d8a-b757-ea73b10c084b.png)

A couple days ago nothing happened like that, but suddenly it happened yesterday. And then i was trying to create a new wordpress project again and error appears like this :
![Screenshot (539)](https://user-images.githubusercontent.com/113582506/226223045-dedb2abd-a9f3-419e-839d-8e3336004160.png)

What should i do to fix it? Please help me. Thank you!

**Comment by Pen-y-Fan** [comment]
These random stoppages can be a number of things, a windows update and false positive on an anti-virus scan, or another program installed, that conflicts.

1. Reboot the computer
2. Check for updates, install and reboot
3. Check your anti-virus program for any malicious activity
4. Has any other software been installed recently? If so try to uninstall it.
5. Check for any conflicts on port 3306 (Laragon > Tools > netstat), and check any usage on port 3306.

It should be **mysqld.exe** in the Laragon directory, EG:

```text
// ...
TCP    0.0.0.0:3306           0.0.0.0:0              LISTENING       5676    F:\laragon\bin\mysql\mysql-8.0.30-winx64\bin\mysqld.exe
// ...
```

**Reply by rahmaazs** [reply]
i've checked my nestat but i don't find the 0.0.0.0:3306 port. these are the items in my nestat:

what should i do? please help me:(

Proto  Local Address          Foreign Address        State           PID     Path
  TCP    0.0.0.0:80             0.0.0.0:0              LISTENING       1008    C:\laragon\bin\apache\httpd-2.4.54-win64-VS16\bin\httpd.exe
  TCP    0.0.0.0:135            0.0.0.0:0              LISTENING       1032
  TCP    0.0.0.0:443            0.0.0.0:0              LISTENING       1008    C:\laragon\bin\apache\httpd-2.4.54-win64-VS16\bin\httpd.exe
  TCP    0.0.0.0:445            0.0.0.0:0              LISTENING       4
  TCP    0.0.0.0:5040           0.0.0.0:0              LISTENING       6048
  TCP    0.0.0.0:49664          0.0.0.0:0              LISTENING       808
  TCP    0.0.0.0:49665          0.0.0.0:0              LISTENING       704
  TCP    0.0.0.0:49666          0.0.0.0:0              LISTENING       1568
  TCP    0.0.0.0:49667          0.0.0.0:0              LISTENING       1352
  TCP    0.0.0.0:49668          0.0.0.0:0              LISTENING       3224
  TCP    0.0.0.0:49669          0.0.0.0:0              LISTENING       776
  TCP    127.0.0.1:83           0.0.0.0:0              LISTENING       4756
  TCP    127.0.0.1:86           0.0.0.0:0              LISTENING       4756
  TCP    127.0.0.1:23402        0.0.0.0:0              LISTENING       1360    C:\Program Files (x86)\NVIDIA Corporation\Update Core\NvBackend.exe
  TCP    192.168.18.109:139     0.0.0.0:0              LISTENING       4
  TCP    192.168.18.109:55448   20.198.119.143:443     ESTABLISHED     3600
  TCP    192.168.18.109:64272   45.121.219.170:443     CLOSE_WAIT      3304    C:\WINDOWS\SystemApps\Microsoft.Windows.Search_cw5n1h2txyewy\SearchApp.exe
  TCP    192.168.18.109:64295   20.205.243.166:443     ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    192.168.18.109:64296   20.205.243.168:443     ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    192.168.18.109:64297   140.82.113.22:443      ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    192.168.18.109:64300   3.92.171.5:443         TIME_WAIT       0
  TCP    192.168.18.109:64304   151.101.40.193:443     ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    192.168.18.109:64306   140.82.112.26:443      ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    192.168.18.109:64310   162.159.135.234:443    ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    [::]:80                [::]:0                 LISTENING       1008    C:\laragon\bin\apache\httpd-2.4.54-win64-VS16\bin\httpd.exe
  TCP    [::]:135               [::]:0                 LISTENING       1032
  TCP    [::]:443               [::]:0                 LISTENING       1008    C:\laragon\bin\apache\httpd-2.4.54-win64-VS16\bin\httpd.exe
  TCP    [::]:445               [::]:0                 LISTENING       4
  TCP    [::]:49664             [::]:0                 LISTENING       808
  TCP    [::]:49665             [::]:0                 LISTENING       704
  TCP    [::]:49666             [::]:0                 LISTENING       1568
  TCP    [::]:49667             [::]:0                 LISTENING       1352
  TCP    [::]:49668             [::]:0                 LISTENING       3224
  TCP    [::]:49669             [::]:0                 LISTENING       776
  TCP    [2404:8000:1015:4d1:d84a:4f0f:67b7:fb76]:55610  [2600:1901:1:e52::]:443  ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    [2404:8000:1015:4d1:d84a:4f0f:67b7:fb76]:55641  [2404:6800:4003:c05::bc]:5228  ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    [2404:8000:1015:4d1:d84a:4f0f:67b7:fb76]:64273  [2600:140e:6:980::1011]:443  CLOSE_WAIT      3304    C:\WINDOWS\SystemApps\Microsoft.Windows.Search_cw5n1h2txyewy\SearchApp.exe
  TCP    [2404:8000:1015:4d1:d84a:4f0f:67b7:fb76]:64274  [2600:140e:6::1720:14cd]:443  CLOSE_WAIT      3304    C:\WINDOWS\SystemApps\Microsoft.Windows.Search_cw5n1h2txyewy\SearchApp.exe
  TCP    [2404:8000:1015:4d1:d84a:4f0f:67b7:fb76]:64275  [2404:8000:10::caa9:2de9]:443  CLOSE_WAIT      3304    C:\WINDOWS\SystemApps\Microsoft.Windows.Search_cw5n1h2txyewy\SearchApp.exe
  TCP    [2404:8000:1015:4d1:d84a:4f0f:67b7:fb76]:64301  [2606:4700::6812:acf]:443  ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    [2404:8000:1015:4d1:d84a:4f0f:67b7:fb76]:64302  [2606:4700::6810:5914]:443  ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    [2404:8000:1015:4d1:d84a:4f0f:67b7:fb76]:64303  [2606:50c0:8002::153]:443  ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  TCP    [2404:8000:1015:4d1:d84a:4f0f:67b7:fb76]:64309  [2606:50c0:8000::154]:443  ESTABLISHED     4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    0.0.0.0:5050           *:*                                    6048
  UDP    0.0.0.0:5353           *:*                                    420     C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    0.0.0.0:5353           *:*                                    420     C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    0.0.0.0:5353           *:*                                    2868
  UDP    0.0.0.0:5355           *:*                                    2868
  UDP    0.0.0.0:52898          *:*                                    4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    0.0.0.0:57629          *:*                                    4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    0.0.0.0:61017          *:*                                    4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    0.0.0.0:61953          *:*                                    4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    0.0.0.0:64041          *:*                                    4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    127.0.0.1:1900         *:*                                    4628
  UDP    127.0.0.1:48300        *:*                                    1360    C:\Program Files (x86)\NVIDIA Corporation\Update Core\NvBackend.exe
  UDP    127.0.0.1:49664        *:*                                    4112
  UDP    127.0.0.1:60872        *:*                                    4628
  UDP    192.168.18.109:137     *:*                                    4
  UDP    192.168.18.109:138     *:*                                    4
  UDP    192.168.18.109:1900    *:*                                    4628
  UDP    192.168.18.109:60871   *:*                                    4628
  UDP    [::]:5353              *:*                                    2868
  UDP    [::]:5353              *:*                                    420     C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    [::]:5355              *:*                                    2868
  UDP    [::]:52898             *:*                                    4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    [::]:57629             *:*                                    4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    [::]:61017             *:*                                    4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    [::]:61953             *:*                                    4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    [::]:64041             *:*                                    4360    C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  UDP    [::1]:1900             *:*                                    4628
  UDP    [::1]:60870            *:*                                    4628
  UDP    [fe80::a293:8fda:f102:ae13%13]:1900  *:*                                    4628
  UDP    [fe80::a293:8fda:f102:ae13%13]:60869  *:*                                    4628

**Reply by Pen-y-Fan** [reply]
Apache is running on ports 80 & 443, however, MySQL isn't running.

Is MySQL showing as started in Laragon, it should look like this:

![MySQL started](https://user-images.githubusercontent.com/40126936/226546044-2573b639-95b4-4531-9c68-d5dd7636e3a7.png)

If so check the MySQL logs: Menu > MySQL > mysqld.log

The log should end:

```text
2023-03-21T07:52:32.714555Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Bind-address: '::' port: 33060
2023-03-21T07:52:32.714784Z 0 [System] [MY-010931] [Server] F:\laragon\bin\mysql\mysql-8.0.30-winx64\bin\mysqld: ready for connections. Version: '8.0.30'  socket: ''  port: 3306  MySQL Community Server - GPL.
```

If there are any errors, google the error and there should be a fix.

Please let us know how you get on.

**Reply by rahmaazs** [reply]
this is when i click start all in laragon:
![image](https://user-images.githubusercontent.com/113582506/226771459-59c196f0-ee52-4935-9c88-12a7d207f8cd.png)

here is mine and it's the same like yours, nothing strange or errors. but i still can't access the laragon.

> 2023-03-22T00:28:03.270965Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Bind-address: '::' port: 33060
2023-03-22T00:28:03.271900Z 0 [System] [MY-010931] [Server] C:\laragon\bin\mysql\mysql-8.0.30-winx64\bin\mysqld: ready for connections. Version: '8.0.30' socket: '' port: 3306 MySQL Community Server - GPL.

i did the telnet localhost 3306 in laragon terminal and it shows that the connection is refused also
![image](https://user-images.githubusercontent.com/113582506/226771786-8da8881c-06f5-4334-8c2b-2950bda1d55f.png)

**Reply by Pen-y-Fan** [reply]
telnet will always return a refused connection, its better to try the MySQL command:

This is the error I received when MySQL wasn't started:

```sh
C:\laragon\www
λ mysql -u root
ERROR 2003 (HY000): Can't connect to MySQL server on 'localhost:3306' (10061)
```

With MySQL started `mysql -u root` works:

```sh
C:\laragon\www
λ mysql -u root
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 8
Server version: 8.0.30 MySQL Community Server - GPL

Copyright (c) 2000, 2022, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql>
```

**Reply by rahmaazs** [reply]
replying about telnet, when i try "telnet localhost 3306" in my other device in Laragon also, it returns connected not refused like i did before (sorry i can't insert the picture)

this is when i try "mysql -u root"
> C:\laragon\www
λ mysql -u root
ERROR 2003 (HY000): Can't connect to MySQL server on 'localhost:3306' (10061)

**Comment by josealgardataborges** [comment]
Please check your windows task manager to check if you have mysqld running:

![image](https://user-images.githubusercontent.com/8929655/227171305-fed048b1-959f-461e-9c78-a803f9fe435b.png)

**Reply by rahmaazs** [reply]
I've checked on it but sadly mysqld isn't running on my windows task manager :(

**Reply by josealgardataborges** [reply]
That's what I suspected, somehow, LARAGON thinks mysql is running, but it's not, hence you cannot connect to it.

The reason unfortunately, is "inside" your own windows instalation, and remotely is very hard to guess what it might be.

Maybe you can try and install a diferent "mysql" version inside the Laragon/bin/mysql directory and change laragon to use that version to see if the problem goes away?

![image](https://user-images.githubusercontent.com/8929655/227242104-d807b0f9-94f3-4c44-b228-6a9dfcb92840.png)

**Reply by rahmaazs** [reply]
I've installed a new mysql version (mysql-8.0.32-winx64) and change the preferences in Laragon but the problem is still there

**Reply by asgharali101** [reply]
yes dear i also run on this problem

**Reply by YaseeraCentric** [reply]
reinstall laragon after backing up the projects

**Reply by wasimmia** [reply]
@rahmaazs Did you fix the issue? I'm also facing the issue

**Reply by domsinclair** [reply]
Similar issue myself.  Mysql is up and running but I get the self same error appearing. The wp-config.php appears to be configured correctly.  May give a full uninstall and reinstall of Laragon a go.  Doesn't help that I have a full version of MSSql running which screws up the default ports that Laragon wants to use.

**Reply by wasimmia** [reply]
I have uninstalled laragon and agian I have installed, but i don't know what was the issue, I have lost my some work

**Reply by domsinclair** [reply]
I reinstalled Laragon (chose a different drive but frankly that shouldn't make a difference). I then had to reset the default port 80 in the preferences for Apache ( as I said before I have MS Sql running so port 80 is problematic).

With that done I clicked the start all button and pretty quicky got an AH00072 error connected with Port 443.  To fix this I navigated to this folder D:\laragon\bin\apache\httpd-2.4.54-win64-VS16\conf\extra (your main drive may be different) opened up the httpd-ss-conf file in note pad and commented out the 'Listen 443' line. Save the file and restarted the servers and both Apache and Mysql came up.

After that I just ran the quick app wordpress menu option (I got a windows security dialog which I accepted) the site was created and when I navigated to it I got the familiar prompts to install wordpress.

**Comment by AlizerUncaged** [comment]
Any fixes for this? This happens every time i attempt to run open a local wordpress

alright so i found the problem, for some reason, every time i restart laragon, mysql keeps resetting the entire database, basically in my config.php in wordpress my database is named "Website", i checked mysql and that database doesnt exist,

**Comment by JoshDanns** [comment]
I kept getting a similar error. 

I fixed the issue by disabling all network connections that were not in use.

Hope this helps.

![image](https://github.com/leokhoa/laragon/assets/37690850/a34e55a6-2abb-49a4-a302-a86640b67f0d)

**Comment by AhmadRaza9** [comment]
![image](https://github.com/leokhoa/laragon/assets/85168169/acc9d45b-8e02-49d7-8054-7d11c2ad62ff)

I'm getting  this error.


I  have copy "laragon" folder from one pc to another
I'm getting "Faild: MySQL error"
And my databases are not showing in "localhost/phpmyadmin"


can you help me please
I'm getting these error, can anyone help me

**Comment by asgharali101** [comment]
Bro whenever you are running WordPress on localhost, you should not edit or
delete any folders while the server is running. You need to stop the server
before deleting or copying files. I recommend using Laragon for WordPress."

**Reply by AhmadRaza9** [reply]
@asgharali101 
I know, we need to stop laragon and then copy paste and delete something
but my issue is another
I think you won't understand my "faild sql" issue

**Reply by asgharali101** [reply]
than you must try laragon 5
